### PR TITLE
delete plugin folder when uninstall plugin

### PR DESCRIPTION
**Pull request by Liaojinghui**
Correspond to https://github.com/neo-project/neo/pull/2696

### CODE REVIEW

File: neo-cli/CLI/MainService.Plugins.cs [hunk 1]
```diff
@@ -174,7 +174,8 @@ private async Task InstallDependenciesAsync(Stream config, HashSet<string> insta
         /// <returns></returns>
         private static bool PluginExists(string pluginName)
         {
-            return File.Exists($"Plugins/{pluginName}.dll");
+            return Directory.Exists($"Plugins/{pluginName}")
+                   && File.Exists($"Plugins/{pluginName}/{pluginName}.dll");
```
Comment (shargon) [inline neo-cli/CLI/MainService.Plugins.cs:178]: If file exists, directory exists

Comment (Liaojinghui) [reply neo-cli/CLI/MainService.Plugins.cs:178]: LOL, Cant argue with that.

Comment (Liaojinghui) [inline neo-cli/CLI/MainService.Plugins.cs:178]:
```suggestion
            return File.Exists($"Plugins/{pluginName}/{pluginName}.dll");
```

**Comment by shargon** [comment]
Tested @superboyiii ?

File: neo-cli/CLI/MainService.Plugins.cs [hunk 2]
```diff
@@ -174,7 +174,7 @@ private async Task InstallDependenciesAsync(Stream config, HashSet<string> insta
         /// <returns></returns>
         private static bool PluginExists(string pluginName)
         {
-            return File.Exists($"Plugins/{pluginName}.dll");
+            return File.Exists($"Plugins/{pluginName}/{pluginName}.dll");
```
Comment (erikzhang) [inline neo-cli/CLI/MainService.Plugins.cs:177]: Changes to

```c#
return Plugin.Plugins.Any(p => p.Name == pluginName);
```

What do you think?

Comment (Liaojinghui) [reply neo-cli/CLI/MainService.Plugins.cs:177]: Also works~~

File: neo-cli/CLI/MainService.Plugins.cs [hunk 3]
```diff
@@ -225,16 +225,12 @@ private void OnUnInstallCommand(string pluginName)
...
             }
-
             try
             {
-                DeleteFiles(new[] { $"Plugins/{pluginName}.dll", $"Plugins/{pluginName}/config.json" });
-                Directory.Delete($"Plugins/{pluginName}", false);
-            }
-            catch (IOException)
-            {
+                DeleteFiles(Directory.GetFiles($"Plugins/{pluginName}"));
```
Comment (erikzhang) [inline neo-cli/CLI/MainService.Plugins.cs:230]: This line is not needed any more, right?

Comment (Liaojinghui) [reply neo-cli/CLI/MainService.Plugins.cs:230]: By setting the recursive of directory delete to `yes`,  this line is unnecessary... but i just consider keeping it here do no harm

**Comment by erikzhang** [comment]
Need to test.

**Comment by Liaojinghui** [comment]
> Need to test.

@nicolegys ~~~ may you please help me test this?

**Comment by nicolegys** [comment]
> > Need to test.
> 
> @nicolegys ~~~ may you please help me test this?

I'll test it.

**Comment by nicolegys** [comment]
> > > Need to test.
> > 
> > 
> > @nicolegys ~~~ may you please help me test this?
> 
> I'll test it.

Sorry, I can't test it now. The master branch of `neo-modules` is incompatible with `neo-3.2.1-CI01365`.

File: neo-cli/CLI/MainService.Plugins.cs [hunk 4]
```diff
@@ -174,7 +174,7 @@ private async Task InstallDependenciesAsync(Stream config, HashSet<string> insta
         /// <returns></returns>
         private static bool PluginExists(string pluginName)
         {
-            return Plugin.Plugins.Any(p => p.Name == pluginName);
+            return Plugin.Plugins.Any(p => p.Name.ToLower() == pluginName);
```
Comment (erikzhang) [inline neo-cli/CLI/MainService.Plugins.cs:177]: `pluginName` doesn't need to be lower case?

Comment (shargon) [reply neo-cli/CLI/MainService.Plugins.cs]: Done

**Comment by Liaojinghui** [comment]
@nicolegys https://github.com/neo-project/neo-modules/pull/713 is merged, we can test this pr now

**Comment by nicolegys** [comment]
plugin can be uninstalled even though others depend on it.

seems something wrong when reading `config.json`
image

**Comment by Liaojinghui** [comment]
Seems the dependency check failed, i will check it out @nicolegys

**Comment by Liaojinghui** [comment]
@nicolegys  fixed

**Comment by erikzhang** [comment]
@nicolegys Can you test it again?

**Comment by nicolegys** [comment]
> plugin can be uninstalled even though others depend on it.  seems something wrong when reading `config.json` image

This issue still exists. @Liaojinghui

**Comment by Liaojinghui** [comment]
@nicolegys that is because you did not use the `correct` plugin name, original plugin name is not ALL lowercases.

**Comment by nicolegys** [comment]
> @nicolegys that is because you did not use the `correct` plugin name, original plugin name is not ALL lowercases.

You are right! But since both `install` and `uninstall` support ignoring case, we'd better maintain consistency.

**Review by nicolegys — APPROVED** [review]
test pass

**Review by erikzhang — APPROVED** [review]

**Pull request merged**